File: java/src/test/java/dojo/supermarket/model/FakeCatalog.java
package dojo.supermarket.model;

import java.util.HashMap;
import java.util.Map;

public class FakeCatalog implements SupermarketCatalog {

    private final Map<String, Product> products = new HashMap<>();
    private final Map<String, Double> prices = new HashMap<>();

    @Override
    public void addProduct(Product product, double price) {
        this.products.put(product.name(), product);
        this.prices.put(product.name(), price);
    }

    @Override
    public double getUnitPrice(Product p) {
        return this.prices.get(p.name());
    }

}

File: kotlin/src/test/kotlin/dojo/supermarket/model/FakeCatalog.kt
package dojo.supermarket.model

import supermarket.model.Product
import supermarket.model.SupermarketCatalog
import java.util.*

class FakeCatalog : SupermarketCatalog {
    private val products = HashMap<String, Product>()
    private val prices = HashMap<String, Double>()

    override fun addProduct(product: Product, price: Double) {
        this.products[product.name] = product
        this.prices[product.name] = price
    }

    override fun getUnitPrice(p: Product): Double {
        return this.prices[p.name]!!
    }
}
